### PR TITLE
Spring Boot intro 1.0

### PR DESCRIPTION
**Pull request by aksoli666**
added checkstyle and the following property;
created a .github directory with workflows and ci.yml; created Book entity;
created BookRepository and BookRepositoryImpl;
created BookService and BookServiceImpl;
added CommandLineRunner bean to main class;

### CODE REVIEW

File: src/main/java/shop/config/AppConfig.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
+import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
+
+@Configuration
+@PropertySource("classpath:application.properties")
+@ComponentScan(basePackages = "shop")
+public class AppConfig {
```
Comment (Elena-Bruyako) [inline src/main/java/shop/config/AppConfig.java]: remove class

File: src/main/java/shop/entity/Book.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Table(name = "books")
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/shop/entity/Book.java]:
```suggestion
@Getter
@Setter
```

File: src/main/java/shop/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import shop.entity.Book;
+import shop.repository.BookRepository;
+
+@Repository
```
Comment (Elena-Bruyako) [inline src/main/java/shop/repository/impl/BookRepositoryImpl.java:14]:
```suggestion
@Repository
@RequiredArgsConstructor
```

File: src/main/java/shop/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/shop/repository/impl/BookRepositoryImpl.java]:
```suggestion
```

File: src/main/java/shop/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new EntityExistsException("Book already exists or something is wrong. "
```
Comment (Elena-Bruyako) [inline src/main/java/shop/repository/impl/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can't save book: " + book, e");
```

File: src/main/java/shop/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        }
+        return book;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("SELECT b FROM Book b", Book.class).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Error finding all books", e);
```
Comment (Elena-Bruyako) [inline src/main/java/shop/repository/impl/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Fetching books failed", e);
```

File: src/main/java/shop/repository/impl/BookRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+                session.close();
+            }
+        }
+        return book;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("SELECT b FROM Book b", Book.class).getResultList();
```
Comment (Elena-Bruyako) [inline src/main/java/shop/repository/impl/BookRepositoryImpl.java]:
```suggestion
            return session.createQuery("FROM Book", Book.class).getResultList();
```

File: src/main/java/shop/service/impl/BookServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
+package shop.service.impl;
+
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import shop.entity.Book;
+import shop.repository.BookRepository;
+import shop.service.BookService;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/shop/service/impl/BookServiceImpl.java:10]:
```suggestion
@Service
@RequiredArgsConstructor
```

File: src/main/java/shop/service/impl/BookServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/shop/service/impl/BookServiceImpl.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by okuzan — APPROVED** [review]

**Pull request merged**